middle_end/flambda2/from_lambda/lambda_to_flambda_primitives.ml

```diff
@@ -1505,114 +1523,117 @@ let convert_lprim ~big_endian (prim : L.primitive) (args : Simple.t list list)
     [ box_float32 mode
         (Unary (Num_conv { src; dst }, unbox_float arg))
         ~current_region ]
-  | Pintoffloat Pfloat64, [[arg]] ->
+  | Pintoffloat Boxed_float64, [[arg]] ->
     let src = K.Standard_int_or_float.Naked_float in
     let dst = K.Standard_int_or_float.Tagged_immediate in
     [Unary (Num_conv { src; dst }, unbox_float arg)]
-  | Pfloatofint (Pfloat64, mode), [[arg]] ->
+  | Pfloatofint (Boxed_float64, mode), [[arg]] ->
     let src = K.Standard_int_or_float.Tagged_immediate in
     let dst = K.Standard_int_or_float.Naked_float in
     [box_float mode (Unary (Num_conv { src; dst }, arg)) ~current_region]
-  | Pnegfloat (Pfloat64, mode), [[arg]] ->
+  | Pnegfloat (Boxed_float64, mode), [[arg]] ->
     [ box_float mode
         (Unary (Float_arith (Float64, Neg), unbox_float arg))
         ~current_region ]
-  | Pabsfloat (Pfloat64, mode), [[arg]] ->
+  | Pabsfloat (Boxed_float64, mode), [[arg]] ->
     [ box_float mode
         (Unary (Float_arith (Float64, Abs), unbox_float arg))
         ~current_region ]
-  | Paddfloat (Pfloat64, mode), [[arg1]; [arg2]] ->
+  | Paddfloat (Boxed_float64, mode), [[arg1]; [arg2]] ->
     [ box_float mode
         (Binary (Float_arith (Float64, Add), unbox_float arg1, unbox_float arg2))
         ~current_region ]
-  | Psubfloat (Pfloat64, mode), [[arg1]; [arg2]] ->
+  | Psubfloat (Boxed_float64, mode), [[arg1]; [arg2]] ->
     [ box_float mode
         (Binary (Float_arith (Float64, Sub), unbox_float arg1, unbox_float arg2))
         ~current_region ]
-  | Pmulfloat (Pfloat64, mode), [[arg1]; [arg2]] ->
+  | Pmulfloat (Boxed_float64, mode), [[arg1]; [arg2]] ->
     [ box_float mode
         (Binary (Float_arith (Float64, Mul), unbox_float arg1, unbox_float arg2))
         ~current_region ]
-  | Pdivfloat (Pfloat64, mode), [[arg1]; [arg2]] ->
+  | Pdivfloat (Boxed_float64, mode), [[arg1]; [arg2]] ->
     [ box_float mode
         (Binary (Float_arith (Float64, Div), unbox_float arg1, unbox_float arg2))
         ~current_region ]
-  | Pfloatcomp (Pfloat64, comp), [[arg1]; [arg2]] ->
+  | Pfloatcomp (Boxed_float64, comp), [[arg1]; [arg2]] ->
     [ tag_int
         (Binary
            ( Float_comp (Float64, Yielding_bool (convert_float_comparison comp)),
              unbox_float arg1,
              unbox_float arg2 )) ]
-  | Punboxed_float_comp (Pfloat64, comp), [[arg1]; [arg2]] ->
+  | Punboxed_float_comp (Unboxed_float64, comp), [[arg1]; [arg2]] ->
     [ tag_int
         (Binary
            ( Float_comp (Float64, Yielding_bool (convert_float_comparison comp)),
              arg1,
              arg2 )) ]
-  | Punbox_float Pfloat64, [[arg]] -> [Unary (Unbox_number Naked_float, arg)]
-  | Pbox_float (Pfloat64, mode), [[arg]] ->
+  | Punbox_float Boxed_float64, [[arg]] ->
+    [Unary (Unbox_number Naked_float, arg)]
+  | Pbox_float (Boxed_float64, mode), [[arg]] ->
```

Comment:
One could argue that this should be Pbox_float (Unboxed_float64, mode), maybe?

Reply:
I struggled with this for a bit, but I wanted to make sure we only apply these operation on values that can be boxed. otherwise we would be able to represent Pbox_int Unboxed_int16, which we don't want to be able to express. And we should keep it identical across int/float/vector